core/authentication_api.php

```diff
@@ -74,6 +74,9 @@
 # @global int $g_cache_current_user_id
 $g_cache_current_user_id = null;
 
+# @global int $g_login_by_bug_token
+$g_login_by_bug_token = false;
```

Comment:
We shouldn't introduce a different authentication model.  We can have a viewer user that is used by this feature, similar to the user used for anonymous access feature.  The name of such user can be configurable, the same way the anonymous access have a config option for username.  Then once the page that supports such tokens validates the token, it calls auth_attempt_script_login( username ), which will login such user for the duration of the request.  This way no other API or changes are necessary for code used by the page.